mlrun/utils/helpers.py

```diff
@@ -987,7 +987,7 @@ def fill_object_hash(object_dict, uid_property_name, tag=""):
     # originated from a datetime, for example), but since this is a one-way dump only for hash calculation,
     # it's valid here.
     data = json.dumps(object_dict, sort_keys=True, default=str).encode()
-    h = hashlib.sha1()
+    h = hashlib.sha256()
```

Comment:
As we spoke offline, this can break compatibility as the hash is used to get functions/artifacts before updating them (in the `store` flows).
Changing the hashing will return a different UID for existing resources, so we might "miss" them and create new resources instead of update.